desc/objectives/utils.py

```diff
@@ -185,11 +170,12 @@ def factorize_linear_constraints(objective, constraint, x_scale="auto"):  # noqa
     # null space & particular solution
     A = A * D[None, unfixed_idx]
     if A.size:
```

Comment:
I will debug the tests but I am confused about this if statement. I assume this is for single constraint vs multiple constraints? What exactly we try to achieve by `A.size`? It should give the total number of elements in A, right? Or no constraint vs with constraints?

Reply:
Yes, I was also confused about that. I feel like without the if statement we should be able to handle both cases. Inside the `qr_xp_null`, we can check if  `A` is rank 0 matrix or empty. If empty return empty `xp`, if rank 0 return 0.

Reply:
If there are no constraints A will be shape (0, dim_x) which means size is 0 but shape[1] is the dimension of x that we want

Reply:
So, A_inv@b for this case is 0, right? To me multiplying empty arrays is very confusing.